modules/pubmaticBidAdapter.js

```diff
@@ -217,6 +289,218 @@ function _checkParamDataType(key, value, datatype) {
+              w: params[key].w || params[key].width || (params[key].sizes ? params[key].sizes[0] : undefined),
+              h: params[key].h || params[key].height || (params[key].sizes ? params[key].sizes[1] : undefined),
+            }
+          };
+          break;
+        case 'sponsoredBy':
+          assetObj = {
+            id: NATIVE_ASSET_ID.SPONSOREDBY,
+            required: params[key].required ? 1 : 0,
+            data: {
+              type: NATIVE_ASSET_DATA_TYPE.sponsored
+            }
+          };
+          break;
+        case 'body':
+          assetObj = {
+            id: NATIVE_ASSET_ID.BODY,
+            required: params[key].required ? 1 : 0,
+            data: {
+              type: NATIVE_ASSET_DATA_TYPE.desc
+            }
+          };
+          break;
+        case 'video':
+          assetObj = {
+            id: NATIVE_ASSET_ID.VIDEO,
+            required: params[key].required ? 1 : 0,
+            video: {
+              minduration: params[key].minduration,
+              maxduration: params[key].maxduration,
+              protocols: params[key].protocols,
+              mimes: params[key].mimes,
+              ext: params[key].ext
+            }
+          };
+          break;
+        case 'ext':
+          assetObj = {
+            id: NATIVE_ASSET_ID.EXT,
+            required: params[key].required ? 1 : 0,
+          };
+          break;
+        case 'data':
+          assetObj = {
+            id: NATIVE_ASSET_ID.DATA,
+            required: params[key].required ? 1 : 0,
+            data: {
+              type: params[key].type,
+              len: params[key].length || params[key].len,
+              ext: params[key].ext
+            }
+          };
+          break;
+        case 'logo':
+          assetObj = {
+            id: NATIVE_ASSET_ID.LOGO,
+            required: params[key].required ? 1 : 0,
+            img: {
+              type: NATIVE_ASSET_IMAGE_TYPE.LOGO,
+              w: params[key].w || params[key].width || (params[key].sizes ? params[key].sizes[0] : undefined),
+              h: params[key].h || params[key].height || (params[key].sizes ? params[key].sizes[1] : undefined)
+            }
+          };
+          break;
+        case 'rating':
+          assetObj = {
+            id: NATIVE_ASSET_ID.RATING,
+            required: params[key].required ? 1 : 0,
+            data: {
+              type: NATIVE_ASSET_DATA_TYPE.rating,
+            }
+          };
+          break;
+        case 'likes':
+          assetObj = {
+            id: NATIVE_ASSET_ID.LIKES,
+            required: params[key].required ? 1 : 0,
+            data: {
+              type: NATIVE_ASSET_DATA_TYPE.likes,
+            }
+          };
+          break;
+        case 'downloads':
+          assetObj = {
+            id: NATIVE_ASSET_ID.DOWNLOADS,
+            required: params[key].required ? 1 : 0,
+            data: {
+              type: NATIVE_ASSET_DATA_TYPE.downloads,
```

Comment:
can we make these all CONSTANTS key names in Upper Case? like NATIVE_ASSET_DATA_TYPE.DOWNLOADS ?

Reply:
Done